src/closures/buoyancy_gradients.jl

```diff
@@ -13,64 +13,87 @@ function buoyancy_gradients(
     param_set::APS,
     bg_model::EnvBuoyGrad{FT, EBG},
 ) where {FT <: Real, EBG <: AbstractEnvBuoyGradClosure}
-
     g = TCP.grav(param_set)
-    molmass_ratio = TCP.molmass_ratio(param_set)
     R_d = TCP.R_d(param_set)
     R_v = TCP.R_v(param_set)
+    T_0 = TCP.T_0(param_set)
+    L_v0 = TCP.LH_v0(param_set)
+    L_s0 = TCP.LH_s0(param_set)
+
+    if bg_model.en_cld_frac < FT(1)
+        T = bg_model.T_unsat
+        Tv_gm = bg_model.Tv_gm
+        ts_unsat = TD.PhaseEquil_pTq(param_set, bg_model.p, T, bg_model.qt_unsat)
+        R_m = TD.gas_constant_air(param_set, ts_unsat)
+        cp_m = TD.cp_m(param_set, ts_unsat)
 
-    phase_part = TD.PhasePartition(FT(0), FT(0), FT(0)) # assuming R_d = R_m
-    Π = TD.exner_given_pressure(param_set, bg_model.p, phase_part)
+        ∂b∂M_unsat = (g / Tv_gm) * (R_m / R_d / cp_m)
+        ∂b∂qt_unsat = (g / Tv_gm) * (R_m / cp_m * (R_d * T_0 - L_v0) + T / R_d * (R_v - R_d))
+    else
+        ∂b∂M_unsat = FT(0)
+        ∂b∂qt_unsat = FT(0)
 
-    ∂b∂θv = g * (R_d * bg_model.ρ / bg_model.p) * Π
+    end
 
-    if bg_model.en_cld_frac > 0.0
-        ts_sat = thermo_state_pθq(param_set, bg_model.p, bg_model.θ_liq_ice_sat, bg_model.qt_sat)
-        phase_part = TD.PhasePartition(param_set, ts_sat)
-        lh = TD.latent_heat_liq_ice(param_set, phase_part)
+    if bg_model.en_cld_frac > FT(0)
+        T = bg_model.T_sat
+        Tv_gm = bg_model.Tv_gm
+        ts_sat = TD.PhaseEquil_pTq(param_set, bg_model.p, T, bg_model.qt_sat)
+        L_v = TD.latent_heat_vapor(param_set, ts_sat)
+        L_s = TD.latent_heat_sublim(param_set, ts_sat)
+        L_f = TD.latent_heat_fusion(param_set, ts_sat)
+        R_m = TD.gas_constant_air(param_set, ts_sat)
         cp_m = TD.cp_m(param_set, ts_sat)
-        ∂b∂θl_sat = (
-            ∂b∂θv * (1 + molmass_ratio * (1 + lh / R_v / bg_model.t_sat) * bg_model.qv_sat - bg_model.qt_sat) /
-            (1 + lh * lh / cp_m / R_v / bg_model.t_sat / bg_model.t_sat * bg_model.qv_sat)
-        )
-        ∂b∂qt_sat = (lh / cp_m / bg_model.t_sat * ∂b∂θl_sat - ∂b∂θv) * bg_model.θ_sat
+        qv_star = TD.vapor_specific_humidity(param_set, ts_sat)
+
+        ∂b∂M_sat = (g / Tv_gm) * (R_m / cp_m - L_v * qv_star / (cp_m * T)) / (1 - L_v^2 * qv_star / (R_v * T^2))
```

Comment:
1/R_d factor missing and signs look wrong:
```suggestion
        ∂b∂M_sat = (g / Tv_gm) / R_d * (R_m / cp_m + L_v * qv_star / (cp_m * T)) / (1 - L_v^2 * qv_star / (R_v * T^2))
```
This is also not quite right yet: You need to distinguish between saturation over liquid and ice, so L_v with liquid and L_s with ice. 

I suggest you define dq_vap_sat_dT in Thermodynamics correctly, with L = lambda * L_v + (1-lambda) * L_s, where lambda is the liquid fraction (given as a function of temperature in equilibrium and as lambda = q_l/(q_i + q_l) otherwise). Then use this dq_vap_sat_dT here consistently. Otherwise, you have to carry the phase partitioning and distinguish between phases every time you use this.